src/ts/typing/open-overlay.d.ts

```diff
@@ -13,11 +13,19 @@ export interface OpenOverlay {
 }
 
 export interface OverlayChangeEvent {
-  type: string;
+  type: EventTypes;
+  /** list of currenly open overlays */
   openOverlays: OpenOverlay[];
   combinedOverlayHeight: number;
+  /** unique id to identify of the overlay holder to identify the chart */
+  overlayHolderId: string;
 }
 
-export interface OverlayChangeSubscriber {
-  (change: OverlayChangeEvent): void;
-}
+/**
+ * @param  {OverlayChangeEvent} change - overlay change event
+ * @param  {SVGSVGElement} chartBaseEl - base chart element of the chart instance that has triggered the update
+ * @returns void
+ */
+export type OverlayChangeSubscriber = (change: OverlayChangeEvent, chartBaseEl: SVGSVGElement) => void;
```

Comment:
I left it there for potential future use, but I think of refactoring this pr and see what the `context` approach would look like.